src/underactuated/pyplot_visualizer.py

```diff
@@ -48,20 +50,56 @@ def __init__(self, draw_timestep=0.033333, facecolor=[1, 1, 1],
 
         self.ax.axis('equal')
         self.ax.axis('off')
+        self.record = False
+        self.recorded_contexts = []
         self.show = (matplotlib.get_backend().lower() != 'template')
-        if self.show:
-            self.fig.show()
+
+        def on_initialize(context, event):
+            if self.show:
+                self.fig.show()
+
+        self.DeclareInitializationEvent(
+            event=PublishEvent(
+                trigger_type=TriggerType.kInitialization,
+                callback=on_initialize))
 
     def DoPublish(self, context, event):
         LeafSystem.DoPublish(self, context, event)
-        self.draw(context)
-        self.fig.canvas.draw()
         if self.show:
+            self.draw(context)
+            self.fig.canvas.draw()
             plt.pause(1e-10)
+        if self.record:
+            snapshot = self.AllocateContext()
+            snapshot.SetTimeStateAndParametersFrom(context)
+            self.FixInputPortsFrom(self, context, snapshot)
+            self.recorded_contexts.append(snapshot)
 
     def draw(self, context):
+        '''' Return a list of updated artists if blitting is used '''
         print "SUBCLASSES MUST IMPLEMENT."
 
+    def start_recording(self, show=True):
+        if not show:
+            self.show = False
```

Comment:
This is super weird looking -- I assume it has to do with the (matplotlib.get_backend().lower() != 'template') logic? A warning for users who leave show=True, but nothing is still showed for that reason, might be nice.